### PR TITLE
ETCD-425: adding etcd tuning profiles enhancement

### PR DESCRIPTION
**Pull request by dusk125**
This enhancement describes allowing customers to tweak etcd settings in a controlled manor.

### CODE REVIEW

File: enhancements/etcd/etcd-tuning-profiles.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
+---
+title: etcd-tuning-profiles
+authors:
+  - "@dusk125"
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - "@hasbro7, etcd team"
+  - "@tjungblu, etcd team"
```
Comment (tjungblu) [inline enhancements/etcd/etcd-tuning-profiles.md:7]: /lgtm

Thanks Allen!

File: enhancements/etcd/etcd-tuning-profiles.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+* Add the currently known profiles:
+  * Default (""):
+    - HEARTBEAT_INTERVAL: Platform dependent
+    - LEADER_ELECTION_TIMEOUT: Platform dependent
+  * Standard:
+    - HEARTBEAT_INTERVAL: 100ms
+    - LEADER_ELECTION_TIMEOUT: 1000ms
+  * "Slower" (or other name):
+    - HEARTBEAT_INTERVAL: 500ms
+    - LEADER_ELECTION_TIMEOUT: 2500ms
```
Comment (hasbro17) [inline enhancements/etcd/etcd-tuning-profiles.md:52]: This should go in the proposal section since this is outlining the actual implementation.
Also can you outline the platform dependent values (or link to the ones we have today) so we have them here for reference on how they differ from the slower/standard profiles.

File: enhancements/etcd/etcd-tuning-profiles.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+The default profile ("" or unset) will allow for upgrades to this feature as it tells the system to choose the values based on the platform: this is the current behavior.
+The values for the two proposed profiles for the Tech Preview of this feature are the current default values (applied to all platform except for) and the values applied for Azure and IBMCloud VPC.
+The latter values have been used successfully in the field for some time so the risk to future cluster is minimal.
+Changing to a "slower" profile will likely incur a performance/latency penalty, but that is likely an acceptable trade for cluster stability.
+
+In this iteration, for the Tech Preview, we will make it clear that changing the profile will require an etcd redeployment.
+In the future enhancement, we can discuss a more seamless transition between profiles.
+We will not allow the user to set arbitrary values for the parameters, they must conform to the profiles values (by way of the profile).
+
+In the Tech Preview, the active profile will be set via the API Server, then an etcd rollout can be triggered to consume the new profile.
```
Comment (hasbro17) [inline enhancements/etcd/etcd-tuning-profiles.md:74]: This makes it sound like we would not have an API after techpreview. But I'm assuming we'll always set the profiles via the opeartor/v1 etcd spec and have etcd rollout to the new latency profile.
So maybe rephrase so this detail isn't specific only to techpreview.

Also can you add a section that outlines the actual API (along with the API's group version and Kind) that we merged in https://github.com/openshift/api/pull/1538/files

That might be good to have here for historical reference otherwise as a first time reader you wouldn't know where to look in openshift/api.
